src/main/java/org/zalando/nakadi/controller/ExceptionHandling.java

```diff
@@ -85,4 +87,18 @@ private String generateErrorTraceId() {
         }
         throw exception.getException();
     }
+
+    @ExceptionHandler(RepositoryProblemException.class)
+    public ResponseEntity<Problem> handleRepositoryProblem(final RepositoryProblemException exception,
+                                                           final NativeWebRequest request) {
+        LOG.error("Repository problem occurred", exception);
+        return Responses.create(Response.Status.SERVICE_UNAVAILABLE, exception.getMessage(), request);
+    }
+
+    @ExceptionHandler(InconsistentStateException.class)
```

Comment:
Maybe it would be better to always log the stacktrace but to the user just send a generic error message: internal server error. To the user it doesn't matter so much if it's the database or a bug in the code. At the end, the only thing the user can do is to retry later.